.github/workflows/helm.yml

```diff
@@ -116,16 +114,19 @@ jobs:
             exit 1
           fi
 
-
       - name: Check changelog
-        if: always()
+        if: ${{ ! inputs.publish-alpha }}
         run: |
           VERSION_PATTERN="([0-9]+\.){2}[0-9]+(-alpha.[0-9][0-9]?)?"
           DATE_PATTERN="[0-9]{4}-[0-9]{2}-[0-9]{2}"
           HEADING_PATTERN="## \[${VERSION_PATTERN}\] - ${DATE_PATTERN}"
           CHANGELOG="${{ needs.set-path.outputs.charts_path }}/CHANGELOG.md"
 
           VERSION=${{ steps.get-chart-version.outputs.CHART_VERSION }}
+          
+          if [[ "${VERSION}" == *-alpha\.* ]]; then
+            exit 0
+          fi
```

Comment:
what are tested here exactly ?

Reply:
Sorry I forgot to describe that in the PR description. 

The idea is to skip the CHANGELOG alpha update so the only file impacted is the `Chart.yaml`.  

On [artifacthub](https://artifacthub.io/packages/helm/substra/substra-backend/25.2.0-alpha.3), the CHANGELOG is never published but linked to the repository's `main` branch (ie. never matching a pre-release) so I felt like this was an heavy and unnecessary manual update. 

WDYT?

Reply:
I agree but you test that with the `if ${{ ! inputs.publish-alpha }}`
Here you want to fail if there is alpha in the changelog ? It's the famous belt + parachute + suspender ? x)

Reply:
With `if ${{ ! inputs.publish-alpha }}` you are skipping the entire step for alpha releases on the manual trigger. 

On the PR trigger, `inputs.publish-alpha` is always `false` so condition is always `true`. It encourages to update the Changelog with the alpha which makes no sense in respect to my previous point.

But now I'm thinking it's actually a good way to block erroneous merge of `alpha` versions into `main` which we have no blocker for... Except the fact that it won't work for the frontend because there's no changelog.

PS: exit 0 means success not failure

Reply:
Actions taken:

* Remove the pre-release segment of the VERSION for CHANGELOG check. If Chart version is `x.y.z-alpha.N`, step will still look for a `x.y.z` entry in the CHANGELOG so no redundant update of file is needed.
* Added step that will automatically fail in a pre-merge (PR) workflow is the Chart version is still an alpha so we don't merge alpha into main

Reply:
Nice catch on the changelog pattern.
And lgtm for the failing step, at least we are SURE that we'll not re-merge any weird chart version... Thanks a lot for taking care of this !